App Submission: noStrudel

hzrd149 opened the pull request:
# App Submission

### noStrudel

### 256x256 SVG icon
https://raw.githubusercontent.com/hzrd149/umbrel-community-app-store/master/hzrd149-nostrudel/icon.svg

### Gallery images
![1](https://raw.githubusercontent.com/hzrd149/nostrudel/v0.37.0/screenshots/profile.png)
![2](https://raw.githubusercontent.com/hzrd149/nostrudel/v0.37.0/screenshots/community.png)
![3](https://raw.githubusercontent.com/hzrd149/nostrudel/v0.37.0/screenshots/photography.png)
![4](https://raw.githubusercontent.com/hzrd149/nostrudel/v0.37.0/screenshots/streaming2.png)

### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [x] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [x] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

nmfretz commented:
Wow, thanks for this submission @hzrd149! This looks great. I will try to review this week.

smolgrrr commented:
Tested on the following instances:

* [x]   x86_64 -> UmbrelOS on proxmox Debian instance
* [x]   Arm64 -> Pi 4 8GB

Works great- I made a small change. We require the digest of the image to be specified so it is pinned until the next update submission.

hzrd149 commented:
bump. whats left to do on this PR? Is there anything I should be doing on my end? thanks

nmfretz commented on nostrudel/umbrel-app.yml:
```diff
@@ -0,0 +1,42 @@
+name: noStrudel
+version: "0.37.1"
+tagline: A sandbox for exploring nostr
+icon: https://raw.githubusercontent.com/hzrd149/umbrel-community-app-store/master/hzrd149-nostrudel/icon.svg
```
This line can be removed completely. We will host the icon here:
https://github.com/getumbrel/umbrel-apps-gallery

hzrd149 replied:
removed :+1:

nmfretz commented on nostrudel/umbrel-app.yml:
```diff
@@ -0,0 +1,42 @@
+  noStrudel is a nostr web client that is focused on exploring the nostr network.
+
+  The focus is on power users or any other users who want to learn the inner workings of nostr.
+releaseNotes: >-
+  Changes:
+
+   - Rebuild DMs view to be more readable
+
+   - Show reactions and zaps on DMs
+
+   - Add support for threads in DMs
+
+   - Add support for Amber signer
+
+   - Add support for NIP-46 signer
+
+   - Support NIP-31 on unknown event kinds
+
```
For this initial launch you can just use quotes here so that no release notes are shown:

```yaml
releaseNotes: ""
```

hzrd149 replied:
updated :+1:

nmfretz commented on nostrudel/umbrel-app.yml:
```diff
@@ -0,0 +1,42 @@
+repo: https://github.com/hzrd149/nostrudel
+support: https://github.com/hzrd149/nostrudel/issues
+port: 3149
+gallery:
+  - https://raw.githubusercontent.com/hzrd149/nostrudel/v0.37.1/screenshots/profile.png
+  - https://raw.githubusercontent.com/hzrd149/nostrudel/v0.37.1/screenshots/community.png
+  - https://raw.githubusercontent.com/hzrd149/nostrudel/v0.37.1/screenshots/photography.png
+  - https://raw.githubusercontent.com/hzrd149/nostrudel/v0.37.1/screenshots/streaming2.png
```
These can be changed to 1.jpg, 2.jpg, etc. For example:

https://github.com/getumbrel/umbrel-apps/blob/f6eae6fdc53a1bd71528fed49118e299da5d1aea/mempool/umbrel-app.yml#L25-L28

We will host the gallery images here as .jpgs:
https://github.com/getumbrel/umbrel-apps-gallery

hzrd149 replied:
updated :+1:

nmfretz reviewed (commented):
Sorry for the delay on my end @hzrd149, this looks great. There are only a few very minor changes I've noted below and then we are ready to launch!

For the gallery images we'll use your screenshots to create four images with text at the top like the other apps have. For example: https://github.com/getumbrel/umbrel-apps-gallery/blob/master/lightning/3.jpg

What text would you like to show at the top of each of your four screenshots? Based on your app description, an example for the first screenshot could be "Explore the Nostr protocol"

hzrd149 commented:
For the text on the screenshots. I think these would work
"Explore the Nostr protocol"
"Manage Communities"
"Watch streams"

nmfretz commented:
Excellent thanks @hzrd149, we will get the gallery images made and then launch the noStrudel!

hzrd149 commented:
Updated this MR with the latest version of the app. I also added a CORS proxy container next to the app in the docker-compose file so the app can use it to get around cors errors when getting metadata or checking nip05 ids

nmfretz commented:
Thanks again for your patience @hzrd149! This is ready to go to the app store. Thank you for the wonderful addition.

Icon and gallery images: https://github.com/getumbrel/umbrel-apps-gallery/tree/master/nostrudel

I bumped to the latest 0.38.3, and added multi-arch digests for the images.

<img width="1383" alt="image" src="https://github.com/getumbrel/umbrel-apps/assets/85373263/a2aed127-15f6-48ec-beb5-c2e226d3e380">
